### ISSUE TITLE
@babel/cli: default the src/out directory

### ISSUE THREAD

**Issue by hzoo** [babel/babel#7237]
https://babeljs.io/docs/usage/cli/#compile-directories

I think we can default `babel` without any arguments to be `babel src --out-dir lib` (default to the src/ directory and output to lib), I guess we may need to check existence and all that?

This is just so if you want to do something in package.json like `"build": "babel src -d lib"`, you can do ``"build": "babel"`` instead.

Can look to other tools that do this kind of thing as well.

> aside: we should add some better error messages around this kind of thing, like if a config isn't found, etc.

**Comment by Andarist** [comment]
`lib` is a good default for me

**Comment by rajzshkr** [comment]
+1 for `lib`

**Comment by xtuc** [comment]
`babel --out-dir mydist` could still be possible? The `src` is by default if you don't specify it.

**Comment by hzoo** [comment]
You can do whatever you want with the flags, this is just so `babel` does what people usually expect.

This + https://github.com/babel/babel/pull/6027 for cli should mean you just have to do `babel`

**Comment by developit** [comment]
Might be worth inspecting what the user has in their `package.json` `"main"` field - it should give the directory you're looking to output to, right?  I believe microbundle and preact-cli use the existence of a `src` directory though, that part is fine just using auto-detection.

**Comment by hzoo** [comment]
Ok so could look to the `main` field folder otherwise default to `lib` or `dist`?

**Comment by iansu** [comment]
I’m interested in doing this if it’s still up for grabs.

**Comment by hzoo** [comment]
Yeah go for it @iansu 👍

**Comment by iansu** [comment]
As I understand it the basic algorithm will be:

1. Try to get the output dir from `main`
2. If not found check for `lib` or `dist` (in which order?)
3. If not found default to `lib` or `dist` (which one?)

The big question is, what's the default `lib` or `dist`?

**Comment by hzoo** [comment]
Yeah, I was thinking `lib` as default. I know `dist` it is for webpack but this is also for `cli` which is used for a library? Since it's probably npmignored I don't know if we can check the folder from init?

**Comment by iansu** [comment]
That logic makes sense to me. I'll use `lib` as the default.

**Comment by nicolo-ribaudo** [comment]
> Since it's probably npmignored

We can check `.npmignore`/`.gitignore`

**Comment by alvaropinot** [comment]
@nicolo-ribaudo that might add some "magic" difficult to reason about by the final user, don't you think?

**Comment by iansu** [comment]
I have an initial implementation of this here: https://github.com/babel/babel/pull/7966

It works but there are still some things to do like updating tests and documentation. Feedback welcome.

**Comment by FernandoLeandroFernandes** [comment]
Sorry for post this here, but I´m facing an issue kinda related with what is being discussed here. I´m trying to transpile my ES6 javascript files in src directory in one go to build directory using th command:
`babel src -d build`
I get the message "Successfully compiled 0 files with Babel.", the directory is created if it doesn't yet exist but no file is compiled (as stated in the happy message). The problem is that is isn't doing the job. :(
